pkg/shell/machines.js

```diff
@@ -161,10 +184,30 @@ define([
             if (machine) {
                 delete machines[host];
                 self.disconnect(host);
-                notify(host, "removed");
+                notify(host);
             }
         }
 
+        self.expect_restart = function expect_restart(host) {
```

Comment:
Even if we do have a more general ```cockpit.expect``` API (noted above) ... then having this as a specific API is still good.